File: truss-compiler/src/main/java/truss/compiler/preprocessor/PreProcessorFileStream.java
package truss.compiler.preprocessor;

import org.antlr.runtime.ANTLRFileStream;

import java.io.IOException;
import java.util.Set;

public class PreProcessorFileStream extends ANTLRFileStream {
    public PreProcessorFileStream(String fileName, Set<String> defines) throws IOException, PreProcessException {
        super(fileName);

        init(defines);
    }

    public PreProcessorFileStream(String fileName, String encoding, Set<String> defines) throws IOException, PreProcessException {
        super(fileName, encoding);

        init(defines);
    }

    private void init(Set<String> defines) throws PreProcessException {
        data = PreProcessor.process(data, n, defines);
        n = data.length;
    }
}

File: Truss.Compiler/PreProcessor/PreProcessorFileStream.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Antlr.Runtime;

namespace Truss.Compiler.PreProcessor {
    public class PreProcessorFileStream : ANTLRFileStream {
        public PreProcessorFileStream(ErrorList errors, string fileName, IEnumerable<string> defines)
            : base(fileName) {
            Init(errors, defines);
        }

        public PreProcessorFileStream(ErrorList errors, string fileName, Encoding encoding, IEnumerable<string> defines)
            : base(fileName, encoding) {
            Init(errors, defines);
        }

        private void Init(ErrorList errors, IEnumerable<string> defines) {
            data = PreProcessor.Process(errors, data, n, defines);
            n = data.Length;
        }
    }
}
